flocker/control/test/test_protocol.py

```diff
@@ -282,13 +298,18 @@ def test_nodestate_updates_node_state(self):
         """
         ``NodeStateCommand`` updates the node state.
         """
+        changes = (NODE_STATE, NONMANIFEST)
         self.successResultOf(
             self.client.callRemote(NodeStateCommand,
-                                   state_changes=(NODE_STATE,),
+                                   state_changes=changes,
                                    eliot_context=TEST_ACTION))
         self.assertEqual(
```

Comment:
:question: Insofar as you're saying both that the same dataset is on the node and that it is not on any node, this test may break someday if we ever add an invariant to enforce that. Probably we won't insofar as state may well be inconsistent due to race conditions, so maybe that's fine.